.github/actions/safety/action.yaml

```diff
@@ -1,24 +1,35 @@
 name: safety
-description: 'Run Safety on frozen dependencies'
+description: 'Run Safety on image'
+inputs:
+  image:
+    description: 'Image name'
+    required: true
+  registry:
+    description: 'Registry to login to pull image, e.g. "ghcr.io" for GHCR, leave empty if image is public'
+    required: false
+  repo_owner:
+    description: 'Name of repository owner, e.g. "github.repository_owner" for ghcr.io'
+    required: false
+  repo_token:
+    description: 'Access token for repository owner, e.g. "secrets.GITHUB_TOKEN" for ghcr.io'
+    required: false
 runs:
   using: "composite"
   steps:
-    - name: Install packages
-      run: |
-        pip3 install --upgrade pip
-        pip3 install -r requirements_dev.txt
-      shell: sh
-    - name: Freeze packages
-      run: pip3 freeze > actual_package_versions.txt
-      shell: sh
-    - name: Install safety
-      run: pip3 install safety
-      shell: sh
+    - name: Login with registry
+      if: inputs.registry != ''
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a # v2.1.0
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ inputs.repo_owner }}
+        password: ${{ inputs.repo_token }}
     - name: Scan
-      run: safety check -r ./actual_package_versions.txt --full-report -o screen --save-json safety-report.json
+      run: |
+        docker run --user 0 -v ${PWD}:/results ${{ inputs.image }} /bin/ash -c \
```

Comment:
do we actually expect different results from the images, that this complicated approach now is necessary?

Reply:
the approach is actually the [recommended one by safety](https://github.com/pyupio/safety/#scan-a-python-based-docker-image). Now, there is 2 reasons to use this one:
- scanning from within the image scans the actually installed versions not the "presumably" installed versions
- scanning from within scans the actual python environment not only the pinned versions from `requirements.txt`
- only in this way can we actually scan the automated builds and the released version separately
that was my rationale for the switch